### PR TITLE
Screening CRUD

### PR DESCRIPTION
**Pull request by koalatosiau**
- added screenings CRUD views and controller
- added form partial

### CODE REVIEW

File: app/controllers/screenings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @screening = Screening.find(params[:id])
+  end
+
+  def set_hall
+    @halls = Hall.all.order(:name)
+  end
+
+  def set_movie
+    @movies = Movie.all.order(:title)
+  end
```
Comment (Protoplaste) [inline app/controllers/screenings_controller.rb:53]: singular form in those names suggest you are setting one hall and one movie
these should be called set_halls and set_movies to indicate you are setting lists of objects

File: app/controllers/screenings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  def destroy
+    @screening.destroy
+
+    redirect_to screenings_path, status: :see_other
+  end
+
+  private
+
+  def set_screening
+    @screening = Screening.find(params[:id])
```
Comment (Protoplaste) [inline app/controllers/screenings_controller.rb:44]: in some of the views you are refering to screenings movie and hall, these will be fetched in separate queries impacting performance
you can fetch everything you need in one query by including those associated object like so:
```suggestion
    @screening = Screening.includes(:movie, :hall).find(params[:id])
```
This is not a major issue with one screening but it's good to keep in mind

File: app/controllers/screenings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+class ScreeningsController < ApplicationController
+  before_action :set_screening, only: %i[show edit update destroy]
+  before_action :set_hall, :set_movie, only: %i[new create edit update]
```
Comment (Protoplaste) [inline app/controllers/screenings_controller.rb:3]: you don't need to set those for create and update actions sience you are not using them there for anything

**Review by Protoplaste — CHANGES_REQUESTED** [review]

File: app/controllers/screenings_controller.rb (hunk 1 above)
Comment (koalatosiau) [reply app/controllers/screenings_controller.rb]: good idea, thank you

**Pull request merged**